src/createServer.js

```diff
@@ -1,3 +1,125 @@
-// Write code here
-// Also, you can create additional files in the src folder
-// and import (require) them here
+
+const http = require('http');
+const { errorMessage } = require('./errorMessage');
+const { convertToCase } = require('./convertToCase/convertToCase');
+
+const supportedCases = ['SNAKE', 'KEBAB', 'CAMEL', 'PASCAL', 'UPPER'];
+
+const createServer = () => {
+  const server = http.createServer((req, res) => {
+    res.setHeader('Content-Type', 'application/json');
+
+    const normalizedUrl = new URL(req.url, `http://${req.headers.host}`);
+    const originalText = normalizedUrl.pathname.slice(1);
+    const targetCase = normalizedUrl.searchParams.get('toCase');
+
+    if (!originalText && !targetCase) {
+      res.statusCode = 400;
+
+      res.write(JSON.stringify({
+        errors: [
+          { message: errorMessage.textIsRequired },
+          { message: errorMessage.caseIsIsRequired },
+        ],
+      }));
+
+      res.end();
+
+      return;
+    }
+
+    if (!originalText && !targetCase) {
+      res.statusCode = 400;
+
+      res.write(JSON.stringify({
+        errors: [
+          { message: errorMessage.textIsRequired },
+          { message: errorMessage.caseIsIsRequired },
+        ],
+      }));
+
+      res.end();
+
+      return;
+    }
+
+    if (!originalText && supportedCases.includes(targetCase)) {
+      res.statusCode = 400;
+
+      res.write(JSON.stringify({
+        errors: [
+          { message: errorMessage.textIsRequired },
+        ],
+      }));
+
+      res.end();
+
+      return;
+    }
+
+    if (!targetCase) {
+      res.statusCode = 400;
+
+      res.write(JSON.stringify({
+        errors: [
+          { message: errorMessage.caseIsIsRequired },
+        ],
+      }));
+
+      res.end();
+
+      return;
+    }
+
+    if (!originalText && !supportedCases.includes(targetCase)) {
+      res.statusCode = 400;
+
+      res.write(JSON.stringify({
+        errors: [
+          { message: errorMessage.textIsRequired },
+          { message: errorMessage.caseIsNotSupported },
+        ],
+      }));
+
+      res.end();
+
+      return;
+    }
+
+    if (!supportedCases.includes(targetCase)) {
+      res.statusCode = 400;
+
+      res.write(JSON.stringify({
+        errors: [
+          { message: errorMessage.caseIsNotSupported },
+        ],
+      }));
+
+      res.end();
+
+      return;
+    }
+
+    res.statusCode = 200;
+
+    const { originalCase, convertedText } = convertToCase(
+      originalText, targetCase,
+    );
+
+    const result = {
+      originalCase,
+      targetCase,
+      originalText,
+      convertedText,
+    };
+
+    res.write(JSON.stringify(result));
+    res.end();
```

Comment:
You can combine this logic and provide JSON to the `end()` function

```suggestion
    res.end(JSON.stringify(result));
```